### ISSUE TITLE
spec "all" invocation policy for shared registrations

### ISSUE THREAD

**Issue by goeddea** [wamp-proto/wamp-proto#202]
what we had in the pre-RFC spec on this issue:
#### "All" Calls

If `CALL.Options.runon == "all"`, the call will be routed to all _Callees_ that registered an implementing endpoint for the called procedure. The calls will run in parallel and asynchronously.

If `CALL.Options.runmode == "gather"` (the default, when `CALL.Options.runmode` is missing), the _Dealer_ will gather the individual results received via `YIELD` messages from _Callees_ into a single list, and return that in `RESULT` to the original _Caller_ - when all results have been received.

If `CALL.Options.runmode == "progressive"`, the _Dealer_ will call each endpoint via a standard `INVOCATION` message and immediately forward individual results received via `YIELD` messages from the _Callees_ as progressive `RESULT` messages (`RESULT.Details.progress == 1`) to the original _Caller_ and send a final `RESULT` message (with empty result) when all individual results have been received.

If any of the individual `INVOCATION`s returns an `ERROR`, the further behavior depends on ..

Fail immediate:

The _Dealer_ will immediately return a `ERROR` message to the _Caller_ with the error from the `ERROR` message of the respective failing invocation. It will further send `INTERRUPT` messages to all _Callees_ for which it not yet has received a response, and ignore any `YIELD` or `ERROR` messages it might receive subsequently for the pending invocations.

The _Dealer_ will accumulate ..

**Comment by oberstet** [comment]
superseded by https://github.com/wamp-proto/wamp-proto/issues/271
